python/python/lance/util.py

```diff
@@ -40,3 +45,71 @@ def sanitize_ts(ts: ts_types) -> datetime:
     elif not isinstance(ts, datetime):
         raise TypeError(f"Unrecognized version timestamp {ts} of type {type(ts)}")
     return ts
+
+
+class KMeans:
```

Comment:
Maybe add a comment or two about the lifecycle of this class.  In other words...

* You need to call fit before you call predict
* Can you call fit twice?

Reply:
Done.

Reply:
Done.